Microsoft.Toolkit.Uwp.UI.Controls/InfiniteCanvas/InfiniteCanvas.cs

```diff
@@ -0,0 +1,289 @@
+
+        private void EnableTextButton_Unchecked(object sender, RoutedEventArgs e)
+        {
+            _canvasTextBox.Visibility = Visibility.Collapsed;
+            _inkCanvas.Visibility = Visibility.Visible;
+            _canvasTextBoxTools.Visibility = Visibility.Collapsed;
+        }
+
+        private void EnableTextButton_Checked(object sender, RoutedEventArgs e)
+        {
+            _inkCanvas.Visibility = Visibility.Collapsed;
+            _canvasTextBoxTools.Visibility = Visibility.Visible;
+        }
+
+        private void EraseAllButton_Click(object sender, RoutedEventArgs e)
+        {
+            _drawingSurfaceRenderer.ClearAll(ViewPort);
+        }
+
+        private async void Current_LeavingBackground(object sender, Windows.ApplicationModel.LeavingBackgroundEventArgs e)
+        {
+            // work around to virtual drawing surface bug.
+            await Task.Delay(1000);
+            _drawingSurfaceRenderer.ReDraw(ViewPort);
+        }
+
+        private void ReDrawCanvas()
+        {
+            _drawingSurfaceRenderer.ReDraw(ViewPort);
+        }
+
+        private void InkScrollViewer_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            ClearTextBoxValue();
+            ReDrawCanvas();
+        }
+
+        private void OnStrokesCollected(InkPresenter sender, InkStrokesCollectedEventArgs args)
+        {
+            _drawingSurfaceRenderer.ExecuteCreateInk(_inkSync.BeginDry());
+            _inkSync.EndDry();
+            ReDrawCanvas();
+        }
+
+        private void UnprocessedInput_PointerMoved(InkUnprocessedInput sender, PointerEventArgs args)
+        {
+            if (_inkCanvasToolBar.ActiveTool == _inkCanvasToolBar.GetToolButton(InkToolbarTool.Eraser))
+            {
+                _drawingSurfaceRenderer.Erase(args.CurrentPoint.Position, ViewPort, _infiniteCanvasScrollViewer.ZoomFactor);
+            }
+        }
+
+        private void InkScrollViewer_ViewChanged(object sender, ScrollViewerViewChangedEventArgs e)
+        {
+            if (!e.IsIntermediate)
+            {
+                ClearTextBoxValue();
+                ReDrawCanvas();
+            }
+        }
+
+        private void DrawingSurfaceRenderer_CommandExecuted(object sender, EventArgs e)
+        {
+            ReRenderCompleted?.Invoke(this, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Redo the last action.
+        /// </summary>
+        public void Redo()
+        {
+            _drawingSurfaceRenderer.Redo(ViewPort);
+        }
+
+        /// <summary>
+        /// Undo the last action.
+        /// </summary>
+        public void Undo()
+        {
+            _drawingSurfaceRenderer.Undo(ViewPort);
+        }
+
+        /// <summary>
+        /// Export the InfinitCanvas as json string.
+        /// </summary>
+        /// <returns>json string</returns>
+        public string ExportAsJson()
+        {
+            return _drawingSurfaceRenderer.GetSerializedList();
```

Comment:
Is there a file missing in the PR, I can't find this implementation?

Reply:
If this file is missing then it won't build or work, _drawingSurfaceRenderer is the engine :)

Reply:
;) Yes, I found the main class, but I didn't see where `GetSerializedList` and its counter-part were implemented.